src/registry.cpp

```diff
@@ -11,14 +11,55 @@
 
 namespace metrics {
 
+namespace {
+
+const auto query_all = [](const tags_t&) -> bool {
+    return true;
+};
+
+struct transformer_t {
+    template<typename T>
+    auto
+    operator()(const std::pair<const tags_t, shared_metric<T>>& pair) const ->
+        std::shared_ptr<tagged_t>
+    {
+        return std::make_shared<shared_metric<T>>(pair.second);
```

Comment:
In general, I don't remember the purpose of shared_metric, instead of shared_ptr<metric_t>, but this one looks weird.